src/ray/raylet/task_spec.cc

```diff
@@ -210,43 +95,57 @@ bool TaskSpecification::IsDriverTask() const {
   return FunctionDescriptor().empty();
 }
 
-Language TaskSpecification::GetLanguage() const {
-  auto message = flatbuffers::GetRoot<TaskInfo>(spec_.data());
-  return message->language();
-}
+rpc::Language TaskSpecification::GetLanguage() const { return message_.language(); }
 
-bool TaskSpecification::IsActorCreationTask() const { return !ActorCreationId().IsNil(); }
+bool TaskSpecification::IsActorCreationTask() const {
+  return message_.type() == rpc::TaskType::ACTOR_CREATION_TASK;
+}
 
-bool TaskSpecification::IsActorTask() const { return !ActorId().IsNil(); }
+bool TaskSpecification::IsActorTask() const {
+  return message_.type() == rpc::TaskType::ACTOR_TASK;
+}
 
 ActorID TaskSpecification::ActorCreationId() const {
-  auto message = flatbuffers::GetRoot<TaskInfo>(spec_.data());
-  return from_flatbuf<ActorID>(*message->actor_creation_id());
+  if (!IsActorCreationTask()) {
+    return ActorID::Nil();
+  }
+  return ActorID::FromBinary(message_.actor_creation_task_spec().actor_id());
 }
 
 ObjectID TaskSpecification::ActorCreationDummyObjectId() const {
-  auto message = flatbuffers::GetRoot<TaskInfo>(spec_.data());
-  return from_flatbuf<ObjectID>(*message->actor_creation_dummy_object_id());
+  if (!IsActorTask()) {
+    return ObjectID::Nil();
+  }
+  return ObjectID::FromBinary(
+      message_.actor_task_spec().actor_creation_dummy_object_id());
 }
 
-int64_t TaskSpecification::MaxActorReconstructions() const {
-  auto message = flatbuffers::GetRoot<TaskInfo>(spec_.data());
-  return message->max_actor_reconstructions();
+uint64_t TaskSpecification::MaxActorReconstructions() const {
+  if (!IsActorCreationTask()) {
+    return 0;
```

Comment:
If we're calling `MaxActorReconstructions` in this case it might be a bug, so maybe this should be a check failure?

Reply:
A similar comment/question applies to some of the other methods here.

Reply:
I tried that, but got a lot of crashes. Can I leave a TODO here and refine that in next PR? As this PR is already very big.

Reply:
@raulchen sure, sounds good.

Reply:
How about using `int64_t` as return type? so that we can return -1 to indicates failure and distinguish it from `0 max_actor_reconstructions` .

Reply:
I'll add a check. then there's no failure case.